subsys/bluetooth/host/keys.c

```diff
@@ -62,17 +94,32 @@ struct bt_keys *bt_keys_get_addr(uint8_t id, const bt_addr_le_t *addr)
 
 #if IS_ENABLED(CONFIG_BT_KEYS_OVERWRITE_OLDEST)
 	if (first_free_slot == ARRAY_SIZE(key_pool)) {
-		struct bt_keys *oldest = &key_pool[0];
+		struct bt_keys *oldest = NULL;
 		bt_addr_le_t oldest_addr;
 
-		for (i = 1; i < ARRAY_SIZE(key_pool); i++) {
+		for (i = 0; i < ARRAY_SIZE(key_pool); i++) {
 			struct bt_keys *current = &key_pool[i];
-
-			if (current->aging_counter < oldest->aging_counter) {
-				oldest = current;
+			bool key_not_in_use = true;
+
+#if (CONFIG_BT_MAX_CONN > 1)
+			key_not_in_use = bt_key_is_not_in_use(i);
+#endif
```

Comment:
I don't think there's any need for the `#if ...` stuff here.  Instead, make the implementation of `bt_key_is_not_in_use()` always available and have it return `true` if MAX_CONN < 2. Then the above just becomes:
```
bool key_not_in_use = bt_key_is_not_in_use();`
```

Reply:
Done